src/directives/column-reorder.js

```diff
@@ -0,0 +1,223 @@
+                        } else if (offset > 0) {
+                            // Insert After
+                            var oldColumn = rowTemplate.splice(oldColIndex, 1)[0];
+                            rowTemplate.splice(newColIndex + 1, 0, oldColumn);
+                        } else {
+                            // Swap Column Places
+                            var oldColumn = rowTemplate.splice(oldColIndex, 1)[0];
+                            var newColumn = rowTemplate.splice(newColIndex, 1)[0];
+                            rowTemplate.splice(newColIndex, 0, oldColumn);
+                            rowTemplate.splice(oldColIndex, 0, newColumn);
+                        }
+
+                        var newRowTemplate = '';
+                        for(var i = 0; i < rowTemplate.length; i++) {
+                            newRowTemplate += rowTemplate[i].outerHTML;
+                        }
+
+                        templateMapper[rowId] = newRowTemplate;
+                    });
+
+                    $scope.$broadcast('drawTableRows');
+                }
+
+            },
+            compile: function($element, $attrs) {
+                var headerRow = $element[0].querySelector('thead tr');
+                headerRow.setAttribute('reorderable-column-template-mapper', getId());
+
+                var headers = headerRow.querySelectorAll('th');
+                var headerCount = headers.length;
+                while (headerCount--) {
+                    headers[headerCount].setAttribute('reorderable-header-drag-drop', '');
+                }
+
+                var bodyRows = $element[0].querySelectorAll('tbody tr');
+                var bodyRowCount = bodyRows.length;
+                while (bodyRowCount--) {
+                    bodyRows[bodyRowCount].setAttribute('reorderable-column-template-mapper', getId());
+                }
+
+            }
+        }
+    });
+
+    module.directive('reorderableColumnTemplateMapper', function($compile) {
+        return {
+            require: '^reorderableColumns',
+            compile: function($element, $attrs) {
+                var directiveName = this.name;
+                var rowId = $attrs[directiveName];
+
+                var uncompiledHtml = $element[0].innerHTML;
+                $element.empty();
+
+                return {
+                    pre: function (scope, element, attrs, ReorderColCtrl) {
+                        ReorderColCtrl.setTemplate(rowId, uncompiledHtml);
+
+                        var buildRow = function() {
+                            element.empty();
+                            var rowTemplate = ReorderColCtrl.getTemplate(rowId);
+                            rowTemplate = angular.element(rowTemplate);
+                            element.append(rowTemplate);
+                            $compile(rowTemplate)(scope);
+                        };
+                        buildRow();
+
+                        scope.$on('drawTableRows', buildRow);
```

Comment:
It should not cast much more than it being compile normally because the original DOM elements were removed so it is not doing a double compile.
The rendering must always be done for all rows, that is how we can get away from not needing to rerender the entire table.
This is only expensive right now because there is no virtual scrolling. Virtual scrolling is what requires us to build the alternate template on render, so that new rows have the correct reordered template and not the template the ng-repeat has stored.